src/lib/orionld/serviceRoutines/orionldPostEntityOperationsUpsert.cpp

```diff
@@ -260,6 +260,9 @@ bool orionldPostEntityOperationsUpsert(ConnectionInfo* ciP)
     // As we will remove items from the tree, we need to save the 'next-pointer' a priori
     // If not, after removing an item, its next pointer point to NULL and the for-loop (if used) is ended
     //
+    int       countIDFields   = 0;
+    int       countTypeFields = 0;
```

Comment:
We don't need any extra variables, we alrewady have a pointer to the ID and TYPE nodes.
Those pointers are set to NULL before the lookup-loop.
So, if before you set the pointers to point to their nodes, you make sure that they are NULL, that's all you need:

```
else if (SCOMPARE5(itemP->name, 't', 'y', 'p', 'e', 0))
{
  if (entityTypeNodeP != NULL)
  {
    ERROR
  }
  entityTypeNodeP = itemP;
}
...
```

We even have a macro for this in `orionld/common/CHECK.h":  DUPLICATE_CHECK

The code snippet from above with DUPLICATE_CHECK:
```
else if (SCOMPARE5(itemP->name, 't', 'y', 'p', 'e', 0))
{
  DUPLICATE_CHECK(entityTypeNodeP, "type", nodeP);
}
```
We also need to make sure that the nodeP is of type JSON string.
So, right before (or after DUPLIOCATE_CHECK):
```
  STRING_CHECK(nodeP, "type");
```

Reply:
FIXED